src/librbd/ImageCtx.cc

```diff
@@ -860,7 +864,7 @@ void _flush_async_operations(ImageCtx *ictx, Context *on_finish) {
 
   void ImageCtx::flush_async_operations() {
     C_SaferCond ctx;
-    _flush_async_operations(this, &ctx);
+    _flush_async_operations(this, new C_AsyncCallback(this, &ctx));
```

Comment:
Is this change necessary?

Reply:
No, that is not necessary.  I thought that all flush-related callbacks be processed by newly added Finisher, with one potential drawback that synchronous flush callbacks might not be delayed since all flush callbacks are pushed  onto the Finisher queue and processed sequentially by the Finisher thread.  So we can leave it unchanged here.